fhd_core/setup_metadata/fhd_struct_init_layout.pro

```diff
@@ -0,0 +1,268 @@
+  ;; earth_degpd is the rotation rate of the earth in degrees per day on the reference date for the array
+  if n_elements(earth_degpd) EQ 0 then earth_degpd=360.985 ;; this is approximately correct
+  ;; ref_date is the reference date on which other parameters apply
+  if n_elements(ref_date) EQ 0 then ref_date='-1' ;; set to something that should be obvious it was a default
+  if n_elements(time_system) EQ 0 then time_system='UTC' ;; assume UTC
+  if n_elements(dut1) EQ 0 then dut1=0 ;; google it. uvfits calls this UT1UTC
+  if n_elements(diff_utc) EQ 0 then diff_utc=0 ;; difference between the time_system and UTC
+  if n_elements(nleap_sec) EQ 0 then begin
+    if time_systemp eq 'IAT' then begin
+      nleap_sec=diff_utc
+    endif else begin
+      nleap_sec=-1 ;; set to something that should be obvious it was a default
+    endelse
+  endif
+  
+  ;; pol_type options are (from AIPS memo 117):
+  ;; ’APPROX’: Linear approximation for circular feeds
+  ;; ’X-Y LIN’: Linear approximation for linear feeds
+  ;; ’ORI-ELP’: Orientation and ellipticity
+  ;; ’VLBI’: VLBI solution form
+  if n_elements(pol_type) EQ 0 then pol_type='X-Y LIN'
+  ;; the antenna table may carry info about polarization characteristics of the feed if known.
+  ;; n_pol_cal_params is the number of such parameters, 0 if no info, 2 if there is info.
+  ;; if there is info, the info is in pola_cal_params & polb_cal_params.
+  if n_elements(n_pol_cal_params) EQ 0 then n_pol_cal_params=0
+  if n_elements(n_antenna) EQ 0 then n_antenna=128
+  
+  
+  if n_elements(antenna_names) EQ 0 then begin
+    antenna_names=string(indgen(n_antenna))
+  endif else begin
+    if n_elements(antenna_names) ne n_antenna then message, 'length of antenna_names must match n_antenna'
+  endelse
+  if n_elements(antenna_numbers) EQ 0 then begin
+    antenna_numbers=indgen(n_antenna)
+  endif else begin
+    if n_elements(antenna_numbers) ne n_antenna then message, 'length of antenna_names must match n_antenna'
+  endelse
+  ;; antenna_coords are relative to the array center in the coordinate_frame
+  if n_elements(antenna_coords) EQ 0 then begin
+    antenna_coords=fltarr(3, n_antenna)
+  endif else begin
+    sz = size(antenna_coords)
+    if sz[0] ne 2 or sz[1] ne 3 or sz[2] ne n_antenna then message, 'antenna_coords must have shape (3, n_antenna)'
+  endelse
+  
+  ;; mount_type codes (from AIPS memo 117): 0 for alt-azimuth, 1 for equatorial, 2 for orbiting, 3 for X-Y,
+  ;; 4 for right-handed Naismith, and 5 for left-handed Naismith are defined. Aperture arrays, which are
+  ;; steered electronically rather than mechanically, are assigned code 6.
+  if n_elements(mount_type) EQ 0 then begin
+    mount_type=0
+  endif else begin
+    if n_elements(mount_type) gt 1 then begin
+      if n_elements(mount_type) ne n_antenna then message, 'mount_type must be a single integer or an array the length of n_antenna'
```

Comment:
Here and in several places below, the `if ... then message, ...` lines get very long. Could you format these with a `$` and new line, like you do for lines 192-193?